### ISSUE TITLE
Is it possible to have Federated Learning on Cloud-Edge?

### ISSUE THREAD

**Issue by Martiniann** [tensorflow/federated#915]
Hi everyone,
 
Currently I am working on a school project about federated learning and came across your framework during exploratory analysis. My project should utilize federated learning in this manner - I have an aggregation server (let's say in a cloud). I want this server to provide model to my 2 Raspberry PIs. These two RPIs would then train the model on a local data for x epochs and provide the trained models/gradients back to the global server. On this server, the results would be federated averaged and new model would  be sent to the PIs. Is such a workflow possible with your framework? If so, could you provide me a hint?
 
Thank you,
Best regards

**Comment by darien-schettler** [comment]
I am also curious about this. Also, if this is not currently possible, what is the estimated timeline for something like this becoming possible within TFF.

**Comment by ESWZY** [comment]
Hey! Same project as me! I'm also working on it.

Would you have any design ideas? Thx!🥰

**Comment by Martin-Stevlik** [comment]
Hello, If you are interested in federated learning I can recommend these 2 frameworks : 
https://github.com/IBM/federated-learning-lib
https://github.com/adap/flower

Good luck with your project :)

**Comment by ESWZY** [comment]
> Hello, If you are interested in federated learning I can recommend these 2 frameworks :
> https://github.com/IBM/federated-learning-lib
> https://github.com/adap/flower
> 
> Good luck with your project :)

Thanks for your kindness help! I will have a try. ❤️

Btw, a good news is I find out a way to implement it based on TFF. 🤣

**Comment by thborba** [comment]
@ESWZY How did you do it? I'm trying to make the server communicate with the client and use its data for training, but no progress so far.

**Comment by ESWZY** [comment]
> @ESWZY How did you do it? I'm trying to make the server communicate with the client and use its data for training, but no progress so far.

It needs time to learn, but you can find some examples here: https://github.com/tensorflow/federated/tree/main/tensorflow_federated/examples

**Comment by ESWZY** [comment]
@thborba Take a look at this example, https://github.com/tensorflow/federated/tree/main/tensorflow_federated/tools

 If you get hello world, it means success
